runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -43,6 +44,13 @@
      */
     final CorfuRuntime runtime;
 
+    /**
+     * The compaction mark of this stream.
+     */
+    @Getter
+    @Setter
+    volatile long compactionMark = Address.NON_ADDRESS;
```

Comment:
I think setCompactionMark should check if it is greater than currentCompactionMark, i.e. Math.max(current, new). Because log unit failure could regress the compactionMark returned by log unit.

Reply:
That is a good point. I have fixed it in next patch.